posthog/hogql/ast.py

```diff
@@ -399,6 +407,18 @@ def print_type(self) -> str:
         return "String"
 
 
+@dataclass(kw_only=True)
+class StringJSONType(StringType):
+    def print_type(self) -> str:
+        return "JSON"
+
+
+@dataclass(kw_only=True)
+class StringArrayType(StringType):
+    def print_type(self) -> str:
+        return "Array"
```

Comment:
This is something extra, happy to move to a separate PR and/or not include this. This makes the "type" on notices be more accurate

<img width="282" height="93" alt="image" src="https://github.com/user-attachments/assets/2aa9fc32-81f2-4f23-838e-7a740e72d83c" />